File: src/bin/aoc1.rb
part1 = File.foreach("input.txt").map do |line|
    line = line.gsub(/[a-z\n]*/, "")
    "#{line[0]}#{line[-1]}".to_i
end.sum

part2 = File.foreach("input.txt").map do |line|
    line = line.gsub("one", "one1one")
               .gsub("two", "two2two")
               .gsub("three", "three3three")
               .gsub("four", "four4four")
               .gsub("five", "five5five")
               .gsub("six", "six6six")
               .gsub("seven", "seven7seven")
               .gsub("eight", "eight8eight")
               .gsub("nine", "nine9nine")
               .gsub("zero", "zero0zero")
               .gsub(/[a-z\n]*/, "")
    "#{line[0]}#{line[-1]}".to_i
end.sum

p "Part 1: #{part1}","Part 2: #{part2}"

File: src/bin/aoc1.rs
use regex::Regex;
use std::fs;

fn main() {
    let input = fs::read_to_string("input.txt").expect("Unable to read file");

    // Part 2
    let input = input
        .lines()
        .map(|g| {
            g.replace("one", "one1one")
                .replace("two", "two2two")
                .replace("three", "three3three")
                .replace("four", "four4four")
                .replace("five", "five5five")
                .replace("six", "six6six")
                .replace("seven", "seven7seven")
                .replace("eight", "eight8eight")
                .replace("nine", "nine9nine")
                .replace("zero", "zero0zero")
        })
        .collect::<Vec<String>>();

    // Part 1
    println!(
        "Result: {:?}",
        input
            .iter()
            .map(|line| {
                let mut itr = line.bytes().filter(|c| c.is_ascii_digit());
                let first = itr.nth(0).unwrap();
                let last = itr.nth_back(0).unwrap_or(first);
                let result = String::from_utf8(vec![first, last]).unwrap();
                result.parse::<i32>().unwrap()
            })
            .sum::<i32>()
    );
}
